tools/azure_controller.sh

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env bash
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+echo $M2_HOME
+echo $PATH
+echo $MAVEN_OPTS
+
+mvn -version
+echo "Commit: $(git rev-parse HEAD)"
+
+# Set up a custom Maven settings file, configuring an Google-hosted maven central
+# mirror.
+cat << EOF > /tmp/az_settings.xml
+<settings>
+  <mirrors>
+    <mirror>
+      <id>google-maven-central</id>
+      <name>GCS Maven Central mirror</name>
+      <url>https://maven-central.storage-download.googleapis.com/repos/central/data/</url>
+      <mirrorOf>central</mirrorOf>
+    </mirror>
+  </mirrors>
+</settings>
+EOF
+
+
+HERE="`dirname \"$0\"`"             # relative
+HERE="`( cd \"$HERE\" && pwd )`"    # absolutized and normalized
+if [ -z "$HERE" ] ; then
+    # error; for some reason, the path is not accessible
+    # to the script (e.g. permissions re-evaled after suid)
+    exit 1  # fail
+fi
+
+source "${HERE}/travis/stage.sh"
+source "${HERE}/travis/shade.sh"
+
+print_system_info() {
+    echo "CPU information"
+    lscpu
+
+    echo "Memory information"
+    cat /proc/meminfo
+
+    echo "Disk information"
+    df -hH
+
+    echo "Running build as"
+    whoami
+}
+
+print_system_info
+
+
+STAGE=$1
+echo "Current stage: \"$STAGE\""
+
+EXIT_CODE=0
+
+#adding -Dmaven.wagon.http.pool=false (see https://developercommunity.visualstudio.com/content/problem/851041/microsoft-hosted-agents-run-into-maven-central-tim.html)
+# --settings /tmp/az_settings.xml 
```

Comment:
?

Reply:
Yeah .. 
The google mirror caused me some nice headache last night :( 
```
2020-02-11T15:40:36.2013808Z [INFO] --- gmavenplus-plugin:1.8.1:execute (merge-categories) @ flink-end-to-end-tests ---
2020-02-11T15:40:36.2032562Z [INFO] Downloading: https://maven-central.storage-download.googleapis.com/repos/central/data/org/codehaus/groovy/groovy-all/2.5.9/groovy-all-2.5.9.pom
2020-02-11T15:40:36.4201087Z [WARNING] The POM for org.codehaus.groovy:groovy-all:pom:2.5.9 is missing, no dependency information available
```
It seems that this file is really not available on the Google mirror. I guess we have to rely on the maven central mirrors :( 

https://dev.azure.com/rmetzger/5bd3ef0a-4359-41af-abca-811b04098d2e/_apis/build/builds/5069/logs/14

Reply:
try clearing the cache, we had the exact same issue on travis a few days ago.

Reply:
I tried that already :( 
Maven does not seem to try different mirrors when something is not available somewhere :(

Reply:
I'll downgrade groovy to 2.5.8, this one seems to be available on the google mirror.

Although it's pretty stupid we have to worry about stuff like this.

Reply:
You can rebase on master or cherry-pick `a984b6a6ca60aa8adbc94397d9dfeee0ac8542fb`.

Reply:
Thanks a lot. I'll rebase.

Reply:
The cherry-picked build now fails with https://maven-central.storage-download.googleapis.com/repos/central/data/org/apache/beam/beam-runners-java-fn-execution/2.19.0/beam-runners-java-fn-execution-2.19.0.pom missing. The artifact has been released on Feb 03.
I just send an email to the guys running the google mirror to see what they have to say :) 

I'm starting to wonder whether we should see if it makes sense to set up our own Maven mirror on one of our build machines: https://www.sonatype.com/download-oss-sonatype

Reply:
Setting up a maven proxy is indeed pretty simple. Got something working locally in a few minutes.